### PR TITLE
[ WIP ] Changes for LLVM 5 release

### PR DESCRIPTION
**Pull request by singam-sanjay**
Changes to compile with LLVM 5.

jitlayers.cpp still needs to be addressed for compile errors.

Will implement version-specific compilation after getting all files to compile and all tests to pass.

### CODE REVIEW

**Comment by singam-sanjay** [comment]
@tobig @StefanKarpinski @yuyichao @tkelman

File: src/cgutils.cpp [hunk 1]
```diff
@@ -2306,7 +2306,7 @@ static void emit_signal_fence(jl_codectx_t &ctx)
     ctx.builder.CreateCall(InlineAsm::get(FunctionType::get(T_void, false), "",
                                       "~{memory}", true));
 #else
-    ctx.builder.CreateFence(AtomicOrdering::SequentiallyConsistent, SingleThread);
+    ctx.builder.CreateFence(AtomicOrdering::SequentiallyConsistent, SyncScope::SingleThread);
```
Comment (singam-sanjay) [inline src/cgutils.cpp]: I'm working on an x86_64 machine. So I'm not sure about the changes required to compile for ARM or AARCH64.

Comment (singam-sanjay) [inline src/cgutils.cpp]: I'm working on an x86_64 machine. So I'm not sure about the changes required to compile for ARM or AARCH64.

File: src/codegen.cpp [hunk 2]
```diff
@@ -1289,8 +1289,11 @@ uint64_t jl_get_llvm_fptr(void *function)
 {
     Function *F = (Function*)function;
     uint64_t addr = getAddressForFunction(F->getName());
-    if (!addr)
-        addr = jl_ExecutionEngine->findUnmangledSymbol(F->getName()).getAddress();
+    if (!addr) {
+	auto expected_addr = jl_ExecutionEngine->findUnmangledSymbol(F->getName()).getAddress();
```
Comment (tkelman) [inline src/codegen.cpp]: spaces, not tabs

Comment (singam-sanjay) [inline src/codegen.cpp]: done.

**Comment by singam-sanjay** [comment]
Could you please help me with the changes for,
1. Passing DebugRegistrar to the object layer in jitlayers.cpp
2. Conflicts in llvm-late-gc-lowering.cpp.

**Comment by yuyichao** [comment]
Replaced by https://github.com/JuliaLang/julia/pull/23034

**Comment by yuyichao** [comment]
Thanks for starting this. There are too many other subtle changes so I started from scratch.

**Comment by singam-sanjay** [comment]
Thank You !

**Pull request closed without merging**